cfme/tests/test_db_restore.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+
+import fauxfactory
+import pytest
+
+from cfme.configure import configuration as config
+from cfme.cloud import provider as cloud_provider
+from cfme.infrastructure.provider import wait_for_a_provider
+from cfme.infrastructure.virtual_machines import Vm
+from fixtures.pytest_store import store
+from utils.appliance import provision_appliance
+from utils.log import logger
+from utils.providers import setup_a_provider
+from utils import testgen, version
+
+pytest_generate_tests = testgen.generate(testgen.provider_by_type, ['virtualcenter'],
```

Comment:
If you implement the second option as I suggest, then the parameter goes completely, which is actually kinda nice, ```pytest_generate_tests``` can go, and the fixture for ```provider_crud```, ```provider_mgmt``` go along with it